echopype/echodata/combine.py

```diff
@@ -79,385 +142,270 @@ def check_and_correct_reversed_time(
+    and the attribute ``reversed_ping_times`` in the
+    ``Provenance`` group will be set to ``1``.
 
     Parameters
     ----------
-    result: EchoData
-        The final ``EchoData`` object representing the combined data
-    old_attrs: Dict[str, List[Dict[str, Any]]]
-        All attributes before combination
-    echodata_filenames : List[str]
-        The source files names for all values in ``echodatas``
-    sonar_model : str
-        The sonar model used for all values in ``echodatas``
+    ed_comb: EchoData
+        ``EchoData`` object that has been constructed from
+        combined ``EchoData`` objects
+    zarr_store: str
+        The zarr store containing the ``ed_comb`` data
+    possible_time_dims: List[str]
+        All possible time dimensions that can occur within
+        ``ed_comb``, which should be checked
+    storage_options: dict
+        Additional keywords to pass to the filesystem class.
 
     Notes
     -----
-    The input ``result`` will be directly modified.
+    If correction is necessary, ``ed_comb`` will be
+    directly modified.
     """
 
-    # store all old attributes
-    for group in old_attrs:
-        all_group_attrs = set()
-        for group_attrs in old_attrs[group]:
-            for attr in group_attrs:
-                all_group_attrs.add(attr)
-        attrs = xr.DataArray(
-            [
-                [group_attrs.get(attr) for attr in all_group_attrs]
-                for group_attrs in old_attrs[group]
-            ],
-            coords={
-                "echodata_filename": echodata_filenames,
-                f"{group}_attr_key": list(all_group_attrs),
-            },
-            dims=["echodata_filename", f"{group}_attr_key"],
-        )
-        result["Provenance"] = result["Provenance"].assign({f"{group}_attrs": attrs})
+    # set Provenance attribute to zero in ed_comb
+    ed_comb["Provenance"].attrs["reversed_ping_times"] = 0
 
-    # Add back sonar model
-    result.sonar_model = sonar_model
+    # set Provenance attribute to zero in zarr (Dataset needed for metadata creation)
+    only_attrs_ds = xr.Dataset(attrs=ed_comb["Provenance"].attrs)
+    only_attrs_ds.to_zarr(zarr_store, group="Provenance", mode="a", storage_options=storage_options)
 
+    for group in ed_comb.group_paths:
 
-def in_memory_combine(
-    echodatas: List[EchoData],
-    sonar_model: str,
-    combine_attrs: str,
-    old_attrs: Dict[str, List[Dict[str, Any]]],
-    old_times: Dict[str, Optional[xr.DataArray]],
-) -> EchoData:
-    """
-    Creates an in-memory (i.e. in RAM) combined ``EchoData``
-    object from the values in ``echodatas``.
+        if group != "Platform/NMEA":
+            # Platform/NMEA is skipped because we found that the times correspond to other
+            # messages besides GPS. This causes multiple times to be out of order and
+            # correcting them is not possible with the current implementation of
+            # _clean_ping_time in qc.api
```

Comment:
```suggestion
            # Platform/NMEA is skipped because we found that the times correspond to other
            # non-GPS messages are often out of order and correcting them is not possible 
            # with the current implementation of _clean_ping_time in qc.api due to excessive recursion.
            # There is also no obvious advantage in correcting the order of these timestamps.
```